packages/wallet-ui/src/slices/walletSlice.ts

```diff
@@ -53,15 +55,19 @@ export const walletSlice = createSlice({
     ) => {
       const accountsToInsert = Array.isArray(payload) ? payload : [payload];
 
-      const accountSet = new Set(state.accounts);
+      const accountSet = new Set(
+        state.accounts.map((account) => account.address),
+      ); // Create a set of addresses
```

Comment:
```suggestion
      const accountSet = new Set(
        state.accounts.map((account) => account.address),
      );
```
this is super obvious it is creating a set of addresses